src/interfaces/phase_integrals.jl

```diff
@@ -0,0 +1,135 @@
+####################
+# The abstract phase integral interface
+#
+# In this file, the abstract interface for different computation methods of
+# phase integrals is defined.
+####################
+"""
+Abstract base type for defining a method for phase integral computation.
+"""
+abstract type ComputeMethod end
+
+"""
+Analytic method for phase integral computation.
+
+Requires an existing implementation of analytic formulas for computing the
+entities in the phase integrals.
+"""
+struct Analytic <: ComputeMethod end
+
+"""
+Fully numerical method for phase integral computation.
+"""
+struct Numeric <: ComputeMethod end
+
+"""
+Struct holding setup specific information to compute phase integrals.
+
+ToDo: We mix physical and numerical aspects in this class.
+This does not seem ideal to me (Klaus).
+"""
+struct PhaseIntegral{F<:AbstractBackgroundField, C<:ComputeMethod}
+    bgfield::F
+    method::C
+
+    function PhaseIntegral(bgfield::F, method::C) where {F<:AbstractBackgroundField, C<:ComputeMethod}
+        _assert_compatibility(bgfield, method)
+        new(bgfield, method)
```

Comment:
Please use explicit returns.
```suggestion
        return new(bgfield, method)
```